tempto-core/src/main/java/com/teradata/tempto/hadoop/FileSystemClient.java

```diff
@@ -45,15 +45,17 @@ InputStream getInputStream()
 
     void createDirectory(String path);
 
+    void deleteDirectory(String path);
+
     void delete(String path);
 
-    void saveFile(String path, InputStream input);
+    void saveFile(String path, InputStream input, long byteLength);
```

Comment:
is this `byteLength` needed? Does it mean we should save first `byteLength` bytes? Maybe we could have an `InputStream` decorator for that instead?